### PR TITLE
Tvlatas/agent oke region lock

### PR DESCRIPTION
**Pull request by agv426**
# Description

This change will lock all jobs running in a pipeline to run in the same region (any AD in the region). The initial region selected to run in will be propagated along to all triggered jobs.

This also will have the OKE cluster run in the same region as the Jenkins runner.

That will limit us to running in regions where we have Jenkins runners. These are the large 3-AD regions: IAD, PHX, LHR, and FRA.

We will not be using the smaller remote regions for OKE clusters after this change.

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: ci/private-registry/Jenkinsfile [hunk 1]
```diff
@@ -28,6 +33,7 @@ pipeline {
     }
 
     parameters {
+        choice (description: 'OCI region to run Jenkins agent and OKE clusters in', name: 'LOCK_REGION', choices: availableRegions)
```
Comment (kminder) [inline ci/private-registry/Jenkinsfile:36]: Should the description indicate the value is actually ignored?

File: tests/e2e/config/scripts/terraform/cluster/create-cluster.sh [hunk 2]
```diff
@@ -29,6 +29,8 @@ done
 # retry 3 times, 30 seconds apart
 tries=0
 MAX_TRIES=3
+export TF_LOG=DEBUG
```
Comment (kminder) [inline tests/e2e/config/scripts/terraform/cluster/create-cluster.sh:32]: Are these intentional?

**Review by kminder — APPROVED** [review]
Approved.

File: ci/private-registry/Jenkinsfile (hunk 1 above)
Comment (agv426) [reply ci/private-registry/Jenkinsfile:36]: Yeah I can update it. Was debating whether to add it now or not. Intention is that eventually we can honor it, when we look at per-region registry support and where the related jobs are run (we want to do this in parallel with other test jobs in the periodics but some release related stuff needs to be worked out)

File: tests/e2e/config/scripts/terraform/cluster/create-cluster.sh (hunk 2 above)
Comment (agv426) [reply tests/e2e/config/scripts/terraform/cluster/create-cluster.sh:32]: yeah, we're tracking down an OKE cluster creation issue that mostly seems to impact OCI integration tests but is intermittent in other tests. The debug helps to tell if it is this issue or something else (and also we need the debug for the service & TF folks)

**Review by stevshap — APPROVED** [review]
refreshing kevin's review

**Pull request merged**